component/src/main/java/me/androidbox/component/agenda/AddVisitorDialog.kt

```diff
@@ -73,11 +76,18 @@ fun AddVisitorDialog(
                 }
             )
 
-            if(!isEmailVerified) {
+            if(!isEmailVerified || isAlreadyAdded) {
+                val errorMessage = if(!isEmailVerified) {
+                    stringResource(id = R.string.email_verify_failed)
+                }
+                else {
+                    stringResource(R.string.visitor_already_added)
+                }
```

Comment:
I would use a **when** here 

```
val errorMessage = when {
    !isEmailVerified -> stringResource(id = R.string.email_verify_failed)
    isAlreadyAdded -> stringResource(R.string.visitor_already_added)
    else -> null
}

if (errorMessage != null) {
// Show the text
}
```
(I'm on phone rn so it might not compile 100%)


Another  approach could be to directly pass the error message to the composable, so you just do the calculation on the viewmodel

Reply:
@MKiperszmid I will check this

Reply:
Updated